programs/util.c

```diff
@@ -693,8 +693,8 @@ static int UTIL_prepareFileList(const char *dirName,
                 ptrdiff_t newListSize = (*bufEnd - *bufStart) + LIST_SIZE_INCREASE;
                 assert(newListSize >= 0);
                 *bufStart = (char*)UTIL_realloc(*bufStart, (size_t)newListSize);
-                *bufEnd = *bufStart + newListSize;
-                if (*bufStart == NULL) { free(path); closedir(dir); return 0; }
+                if (*bufStart != NULL) *bufEnd = *bufStart + newListSize;
+                else { free(path); closedir(dir); return 0; }
```

Comment:
I think this is a useful change since performing arithmetic on a NULL pointer is UB. Can you please add curly braces around `*bufEnd = *bufStart + newListSize;`?